app/grandchallenge/components/tasks.py

```diff
@@ -961,6 +963,50 @@ def add_images_to_component_interface_value(
     civ.image.update_viewer_groups_permissions()
 
 
+@shared_task
+def add_file_to_component_interface_value(
+    *,
+    component_interface_value_pk,
+    user_upload_pk,
+    target_pk,
+    target_app,
+    target_model,
+):
+    user_upload = UserUpload.objects.get(pk=user_upload_pk)
+    civ = get_model_instance(
+        pk=component_interface_value_pk,
+        app_label="components",
+        model_name="componentinterfacevalue",
+    )
+    target = get_model_instance(
+        pk=target_pk,
+        app_label=target_app,
+        model_name=target_model,
+    )
+    error = None
+    with transaction.atomic():
+        try:
+            civ.validate_user_upload(user_upload)
+            civ.full_clean()
+        except ValidationError as e:
+            transaction.set_rollback(True)
+            error = str(e)
+        else:
+            user_upload.copy_object(to_field=civ.file)
+            civ.save()
+
+    if error is not None:
+        Notification.send(
+            type=NotificationType.NotificationTypeChoices.FILE_COPY_STATUS,
+            actor=user_upload.creator,
+            message=f"File for interface {civ.interface.title} failed validation.",
+            target=target,
+            description=(
+                f"File for interface {civ.interface.title} failed validation:\n{error}."
+            ),
+        )
+
```

Comment:
For archive items, I think we delete the user upload object after its file contents have been moved to the CIV field.